fe/src/app.tsx

```diff
@@ -1,27 +1,44 @@
 import Router, { RouterOnChangeArgs } from "preact-router";
-import { HomePage } from "./pages/home.page";
+import Redirect from "./utils/redirect";
+import "fe-utils";
+import { PageWrapper, PageProps } from "./utils/page-wrapper";
 import { FallbackPage } from "./pages/fallback.page";
 import Match from "preact-router/match";
-import "fe-utils";
-import { HeaderView } from "./views/header.view";
-import { PeoplePage } from "./pages/people.page";
+import { HomePage } from "./pages/home.page";
 import { WorldPage } from "./pages/world.page";
-import { SelvesPage } from "./pages/selves.page";
-import { SearchPage } from "./pages/search.page";
+import { ArticleListPage } from "./pages/article-list.page";
 import { PeopleDetailPage } from "./pages/people-detail.page";
+import { SelvesPage } from "./pages/selves.page";
 
 export function App() {
   return (
     <>
-      {window.location.pathname !== "/" && <HeaderView />}
       <div style={{ viewTransitionName: "page" }}>
         <Router>
-          <Match path="/">{() => <HomePage />}</Match>
-          <Match path="/people">{() => <PeoplePage />}</Match>
-          <Match path="/people/:id">{() => <PeopleDetailPage />}</Match>
-          <Match path="/world">{() => <WorldPage />}</Match>
-          <Match path="/about-us">{() => <SelvesPage />}</Match>
-          <Match path="/search">{() => <SearchPage />}</Match>
+          <Redirect path="/" to="/en" />
+          <PageWrapper path="/:lang">
+            {(props: PageProps) => <HomePage {...props} />}
+          </PageWrapper>
+          <PageWrapper path="/:lang/world">
+            {(props: PageProps) => <WorldPage {...props} />}
+          </PageWrapper>
+          <PageWrapper path="/:lang/world/:area">
+            {(props: PageProps) => <WorldPage {...props} />}
+          </PageWrapper>
+          <PageWrapper path="/:lang/article">
+            {(props: PageProps) => <ArticleListPage {...props} />}
+          </PageWrapper>
+          <PageWrapper path="/:lang/article/:locationId/:articleId">
+            {(props: PageProps) => <PeopleDetailPage {...props} />}
+          </PageWrapper>
+          <PageWrapper path="/:lang/selves">
+            {(props: PageProps) => <SelvesPage {...props} />}
+          </PageWrapper>
+          <Match path="/404">
```

Comment:
Is this path 404 match necessary? If we remove code from line 37-41 it will still fall into default fall back page if I am not mistaking.

Reply:
It has not to be /404, or it could be come with parameters later on. Currently, it is more like a route for "catched error". This route will only be redirect from our code. The fallback route is basically for dev purposes, which does not redirect, in case of the page should be there but not yet implemented. The fallback route should be removed or redirect to 404 later. So if we want, later, we can use to something like "/404?reason=Language%20is%20not%set" so that this page can show a specific error.